### PR TITLE
Import autoinstall docs from discourse and add ability to run link substitutions before exporting

### PR DESCRIPTION
**Pull request by ogayot**
This PR imports to the tree the following documents as markdown:

* https://discourse.ubuntu.com/t/automated-server-installs/16612
* https://discourse.ubuntu.com/t/automated-server-install-reference/16613
* https://discourse.ubuntu.com/t/automated-server-install-quickstart/16614
* https://discourse.ubuntu.com/t/automated-server-install-schema/16615

The files have cross references to each other using relative links (e.g., `href=autoinstall-reference.md`).
When building the documentation locally, the links are automatically replaced with the `.html` equivalent.
When building the documentation with the intent to export it to discourse, the links are automatically replaced with their actual location on discourse.

If one wants to add a new documentation file, he or she should update `substitute-links.py` and set the proper substitutions.

To build the documentation locally, use `make` or `make html`.
To build the documentation before exporting it to discourse, use `make discourse`.

Otherwise, the documentation files are brows-able on Github.

### CODE REVIEW

File: documentation/Makefile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+SRC = $(wildcard *.md)
+
+DISCOURSE_MD = $(addprefix discourse/, $(SRC))
+HTML_MD = $(addprefix html/, $(SRC))
+HTML = $(HTML_MD:.md=.html)
+
+discourse/%.md: %.md
+	@mkdir --parents discourse
```
Comment (dbungert) [inline documentation/Makefile:8]: gitignore entries for these directories would be nice

**Review by dbungert — APPROVED** [review]
Nice work.
I have some minor suggestions then please proceed.

File: documentation/Makefile (hunk 1 above)
Comment (ogayot) [reply documentation/Makefile:8]: Good point! Added.

File: documentation/Makefile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	python3 ./substitute-links.py md-to-discourse --input $^ --output $@
+
+html/%.md: %.md
+	@mkdir --parents html
+	python3 ./substitute-links.py md-to-html --input $^ --output $@
+
+%.html: %.md
+	pandoc $^ --standalone --output $@
```
[the remark this answers is not in the document]
Comment (ogayot) [reply documentation/Makefile:16]: Added thanks. Tested in a container and pandoc seems to be the only dependency.

**Pull request merged**